src/app/design/styled-button/styled-button.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Input, OnInit} from '@angular/core';
+@Component({
+  selector: 'app-styled-button',
+  templateUrl: './styled-button.component.html',
+  styleUrls: ['./styled-button.component.scss']
+})
+export class StyledButtonComponent implements OnInit {
+
+  @Input() disabled = false;
+  @Input() styleClass: string;
+  ngOnInit() {
+
+  }
+
+  @Input() onClick: (event: Event) => void = () => {};
```

Comment:
If you want to pass data to a parent use a [EventEmitter](https://angular.io/api/core/EventEmitter#usage-notes).

I would do it: `<app-styled-button (click)="...">`. @MaxiHuHe04 is there a problem if the click event is not on the button but on the component?